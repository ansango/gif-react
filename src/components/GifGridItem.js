import PropTypes from "prop-types";

const GifGridItem = ({ url, title }) => {
  return (
    <li className="border-2 border-gray-100 rounded-lg flex flex-col justify-between">
      <img
        src={url}
        alt={title}
        className="object-fill h-48 w-full rounded-t-lg"
      />

      <p className="p-5">{title}</p>
    </li>
  );
};

GifGridItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default GifGridItem;
