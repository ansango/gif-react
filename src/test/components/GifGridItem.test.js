import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Test <GifGridItem/>", () => {
  const title = "Titulo";
  const url = "http://";

  test("should render component", () => {
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    expect(wrapper).toMatchSnapshot();
  });
});
