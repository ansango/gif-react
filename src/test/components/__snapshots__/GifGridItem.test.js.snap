// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem/> should render component 1`] = `
<li
  className="border-2 border-gray-100 rounded-lg flex flex-col justify-between"
>
  <img
    alt="Titulo"
    className="object-fill h-48 w-full rounded-t-lg"
    src="http://"
  />
  <p
    className="p-5"
  >
    Titulo
  </p>
</li>
`;
